### PR TITLE
[CD] Adds python docker pipeline

### PR DESCRIPTION
**Pull request by perdasilva**
## Description ##
Updates CD pipeline to post python docker images. Addressed comments from closed PR #16435 
@marcoabreu I've decided I would like to finish this. I've addressed your previous comments.

[Example CI-dev run](http://jenkins.mxnet-ci-dev.amazon-ml.com/blue/organizations/jenkins/restricted-mxnet-cd%2Fmxnet-cd-release-job/detail/mxnet-cd-release-job/312/pipeline)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: cd/utils/mxnet_base_image.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+mxnet_variant=${1:?"Please specify the mxnet variant as the first parameter"}
+
+case ${mxnet_variant} in 
+    cu80*)
+    echo "nvidia/cuda:8.0-cudnn7-runtime-ubuntu16.04"
+    ;;
+    cu90*)
+    echo "nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04"
+    ;;
+    cu91*)
```
Comment (aaronmarkham) [inline cd/utils/mxnet_base_image.sh]: I always pick on this one... but isn't it not supported?

File: ci/build.py [hunk 2]
```diff
@@ -27,63 +27,17 @@
...
-from util import *
-import docker
-import docker.models
-import docker.errors
-import signal
-import atexit
-import pprint
-
-
-class Cleanup:
```
Comment (aaronmarkham) [inline ci/build.py]: So all of this is in the new docker client?

File: ci/build.py [hunk 3]
```diff
@@ -305,61 +246,7 @@ def container_run(platform: str,
                     {'bind': '/work/ccache', 'mode': 'rw'},
             },
             environment=environment)
-        try:
-            logging.info("Started container: %s", trim_container_id(container.id))
-            # Race condition:
```
Comment (aaronmarkham) [inline ci/build.py]: So this is no longer an issue?

File: cd/utils/mxnet_base_image.sh (hunk 1 above)
Comment (perdasilva) [reply cd/utils/mxnet_base_image.sh]: Good catch! I'll remove

File: ci/build.py (hunk 2 above)
Comment (perdasilva) [reply ci/build.py]: Yup - it all moved to the "safe_docker_run.py" script

File: ci/build.py (hunk 3 above)
Comment (perdasilva) [reply ci/build.py]: It's still a thing - the signal blocking has moved to the "safe_docker_run.py" script ^^

Comment (perdasilva) [reply ci/build.py]: I'll move these comments in there for clarity though - good catch!

**Comment by perdasilva** [comment]
@aaronmarkham you can see the [push statement](http://jenkins.mxnet-ci-dev.amazon-ml.com/blue/organizations/jenkins/restricted-mxnet-cd%2Fmxnet-cd-release-job/detail/mxnet-cd-release-job/312/pipeline/614#step-627-log-102) - it's not possible to put a link to the image (the best that can be done is a link to the [repository](https://hub.docker.com/r/mxnetdev/python/tags). I'll add a message with the docker pull command to run to get it. Let me know if you would prefer something else =)

**Comment by aaronmarkham** [comment]
Just notice this:
`"SecretString":"{\\"username\\": \\"mxnetcddev\\", \\"password\\": \\"uuz:C^P;2lOR3Z:mNi|:8A6^}UUKtq9F\\"}"`
And the security token is output in the log too... maybe they're one time use? Or?...

**Comment by perdasilva** [comment]
@aaronmarkham nope, that's bad. I've rotated the credentials. I've checked the DockerHub account and nothing has changed there. Thanks for catching this! I've updated the logging configuration and I'm running a job to double check.

**Comment by perdasilva** [comment]
Okay, [looks good](http://jenkins.mxnet-ci-dev.amazon-ml.com/blue/organizations/jenkins/restricted-mxnet-cd%2Fmxnet-cd-release-job/detail/mxnet-cd-release-job/320/pipeline) now. I'm.glad you took the time to have a look! Thank you!!

**Comment by perdasilva** [comment]
@aaronmarkham if you are cool with everything, please merge when you get a chance =D

**Pull request merged**